src/messages/MClientRequest.h

```diff
@@ -161,6 +161,45 @@ class MClientRequest final : public MMDSOp {
     return get_flags() & CEPH_MDS_FLAG_ASYNC;
   }
 
+  std::pair<unsigned,unsigned> get_owner_uid_gid() const {
+    unsigned uid = 0;
+    unsigned gid = 0;
+
+    if (head.version < 3)
```

Comment:
This is incorrect. You should do the version check when `encoding/decoding` the `head` message in https://github.com/ceph/ceph/blob/main/src/include/ceph_fs.h#L659-L707C1.

Reply:
Yes, I've seen that there is some version checking logic in encoding/decoding, but at the same time in this code we do nothing with `h.args` (`bl.append((char*)&h.args, sizeof(h.args));`), but only with handling direct members of `struct ceph_mds_request_head`.

At the same time we have this in the current code:
https://github.com/ceph/ceph/blob/8c4d93c2d72698060289579a704fac9bae07da41/src/messages/MClientRequest.h#L130

I'm ready to rework it as you propose, but please confirm.

Reply:
Please fix this together with https://github.com/ceph/ceph/pull/52575#discussion_r1271622835.

Reply:
Have done!